src/client/components/presetsDropdownSelector/presetsDropdownSelector.js

```diff
@@ -1,14 +1,63 @@
-import React from 'react';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import fetchRequest from '../../helpers/fetchRequest';
+import { HOST } from '../../helpers/constants';
+import { setPresetValue } from '../../store/actions/blocksActions';
+import { addNewPresetsFromDB } from '../../store/actions/presetsAction';
+
 import './presetsDropdownSelector.css';
 
-const PresetsDropdownSelector = () => {
-  const values = ['Jazz', 'Rock', 'Rap'];
+class PresetsDropdownSelector extends Component {
+  componentDidMount() {
+    const { addNewPresetsFromDB } = this.props;
+    fetchRequest.get(`${HOST}/presets`)
+      .then(response => addNewPresetsFromDB(response.data.userPresets));
+  }
+
+  componentDidUpdate(prevProps) {
+    const { blocksData, audioData } = this.props;
+    if (audioData.trackName !== null && blocksData !== prevProps.blocksData) {
+      const prevBlocksData = prevProps.blocksData;
+      prevBlocksData.forEach(({ createEffect, isVisible }) => {
+        isVisible && audioData.sound.removeEffect(createEffect);
+      });
+      blocksData.forEach(({ createEffect, isVisible }) => {
+        isVisible && audioData.sound.addEffect(createEffect);
+      });
+    }
+  }
+
+  handleSelectorChange = (event) => {
+    const { setPresetValue } = this.props;
+    setPresetValue(event.target.value);
+  };
+
+  render() {
+    const { presetsData } = this.props;
+    return (
+      <select className="SlidersComponent__header--selector" onChange={this.handleSelectorChange}>
+        {presetsData.map((value, i) => <option className="option" key={i} value={value}>{value}</option>)}
```

Comment:
You could move this mapping to a separate function

Reply:
done)